### ISSUE TITLE
Can't change parameters of type xsd:dateTime

### ISSUE THREAD

**Issue by Ragnar87** [zaidka/genieacs#244]
The sanitizeParameterValue function in device.js is causing a failure when I am trying to change parameters of type, xsd:dateTime

```
case 'xsd:dateTime':
        if (typeof parameterValue[0] !== 'number') {
          let v = +parameterValue[0];
          if (isNaN(v)) {
            v = Date.parse(parameterValue[0]);
            if (isNaN(v))
              v = '' + parameterValue[0];
          }
          parameterValue[0] = v;
        }
        break;

```
It seems to be the v = Date.parse(parameterValue[0]); line that messes up the format.

I tried to change "InternetGatewayDevice.ManagementServer.PeriodicInformTime" to "0001-01-01T00:03:00.000Z", which should be a valid format, however after being sanitized genieACS sends "-62135596620000" to the cpe, which causes an error. 

If i send "0001-01-01T00:03:00.000Z" to the cpe without sanitizing it, it is accepted

**Comment by zaidka** [comment]
Can you confirm the device reports the param type as "xsd:dateTime"?

**Comment by Ragnar87** [comment]
I am not sure what you mean, but if you are suggesting that the datamodel might be incorrectly configured on the cpe, I am not sure how to verify this. However I can verify that Genieacs has the parameter stored as "xsd:dateTime" through the api. Also if parameterValue is not sanitized, (I did this by commenting out `parameterValue[0] = v;`) the cpe has no problem setting the parameter.

The following is a packet capture between the server and the cpe:
From ACS to the cpe:

```
<soap-env:Body>
    <cwmp:SetParameterValues>
        <ParameterList
            soap-enc:arrayType="cwmp:ParameterValueStruct[1]">
            <ParameterValueStruct>
                <Name>
                    InternetGatewayDevice.X_000E50_TimeOfDay.Schedule.1.TimeOut.ExpireTime
                    </Name>
                <Value
                    xsi:type="xsd:dateTime">
                    -62135593200000
                    </Value>
                </ParameterValueStruct>
            </ParameterList>
        <ParameterKey/>
        </cwmp:SetParameterValues>
</soap-env:Body>
```

From the CPE back to the ACS:
```
<soapenv:Body>
    <soapenv:Fault>
        <faultcode>
            Client
            </faultcode>
        <faultstring>
            CWMP fault
            </faultstring>
        <detail>
            <cwmp:Fault>
                <FaultCode>
                    9003
                    </FaultCode>
                <FaultString>
                    Invalid arguments
                    </FaultString>
                <SetParameterValuesFault>
                    <ParameterName>
                        InternetGatewayDevice.X_000E50_TimeOfDay.Schedule.1.TimeOut.ExpireTime
                        </ParameterName>
                    <FaultCode>
                        9007
                        </FaultCode>
                    <FaultString>
                        Invalid parameter value
                        </FaultString>
                    </SetParameterValuesFault>
                </cwmp:Fault>
            </detail>
        </soapenv:Fault>
    </soapenv:Body>
```

In the example above I tried setting another dateTime parameter to see if the same error occurred, as it did I am guessing the problem is the same for all dateTime parameters.

I hope this helps :-)

**Comment by zaidka** [comment]
I'm not able to reproduce the problem using the simulator. What commit are you on?

**Comment by Ragnar87** [comment]
git describe --tags
v1.1.1

git rev-parse --short HEAD
843d183

I am using node v7.7.4 I will try and change it to either 8.* or 6.* as you have recommended in the readme, I will report back if this changes anything.

I can mention that the devices I have used are Technicolor and FRITZ!Box, and neither worked. The Technicolor returned an error, but the Fritz!Box didn't. From the cwmp perspective the update seemed to work fine, and the parameter was updated, however when I refreshed the parameter again to see what the value was on the cpe, it was 0000-12-31T23:59:59.000Z, which is not what I tried to set it to.

The problem as I see it, is that the function changes a valid dateTime to a negative integer in these lines:
first
`v = Date.parse(parameterValue[0]);`
Which from my understanding, returns the difference between the date provided and January 1, 1970, 00:00:00 UTC in milliseconds or NAN if the date is invalid.
second
`parameterValue[0] = v;`
which sets the parameter to the new value.

Is it possible that:
```
            if (isNaN(v))
              v = '' + parameterValue[0];
```
should be:
```
            if (!isNaN(v))
              v = '' + parameterValue[0];
```
so that if the date is valid (which it would be for Date.parse) to return an integer, v is set to the original parameter value?
